pinot-core/src/main/java/com/linkedin/pinot/core/plan/maker/PlanNodeFactory.java

```diff
@@ -0,0 +1,161 @@
+    }
+    return true;
+  }
+
+  private static boolean isMetadataBasedAggregationFunction(AggregationInfo aggregationInfo,
+      IndexSegment indexSegment) {
+    String aggregationType = aggregationInfo.getAggregationType();
+    if (aggregationType.equalsIgnoreCase(AggregationFunctionFactory.AggregationFunctionType.COUNT.getName())) {
+      return true;
+    }
+
+    // Use minValue and maxValue from metadata, in non star tree cases
+    // minValue and maxValue is generated from dictionary on segment load, so it won't be correct in case of star tree
+    if ((aggregationType.equalsIgnoreCase(AggregationFunctionFactory.AggregationFunctionType.MAX.getName())
+        || aggregationType.equalsIgnoreCase(AggregationFunctionFactory.AggregationFunctionType.MIN.getName())
+        || aggregationType.equalsIgnoreCase(AggregationFunctionFactory.AggregationFunctionType.MINMAXRANGE.getName()))
+        && !indexSegment.getSegmentMetadata().hasStarTree()) {
+      String column = aggregationInfo.getAggregationParams().get("column");
+      SegmentMetadataImpl segmentMetadata = (SegmentMetadataImpl) indexSegment.getSegmentMetadata();
+      if (segmentMetadata.getColumnMetadataMap() == null) {
+        return false;
+      }
+      ColumnMetadata columnMetadata = segmentMetadata.getColumnMetadataFor(column);
+      if (columnMetadata.getMaxValue() != null && columnMetadata.getMinValue() != null) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  /**
+   * Helper method to identify if query is fit to be be served purely based on dictionary.
+   * It can be served through dictionary only for min, max, minmaxrange queries as of now,
+   * and if a dictionary is present for the column
+   * @param brokerRequest Broker request
+   * @param indexSegment
+   * @return True if query can be served using dictionary, false otherwise.
+   */
+  public static boolean isFitForDictionaryBasedPlan(BrokerRequest brokerRequest,
+      IndexSegment indexSegment) {
+    // Skipping dictionary in case of star tree. Results from dictionary won't be correct
+    // because of aggregated values in metrics, and ALL value in dimension
+    if ((brokerRequest.getFilterQuery() != null) || brokerRequest.isSetGroupBy()
+        || indexSegment.getSegmentMetadata().hasStarTree()) {
+      return false;
+    }
+    List<AggregationInfo> aggregationsInfo = brokerRequest.getAggregationsInfo();
+    if (aggregationsInfo == null) {
+      return false;
+    }
+    for (AggregationInfo aggregationInfo : aggregationsInfo) {
+      if (!isDictionaryBasedAggregationFunction(aggregationInfo, indexSegment)) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  private static boolean isDictionaryBasedAggregationFunction(AggregationInfo aggregationInfo,
+      IndexSegment indexSegment) {
+    String aggregationType = aggregationInfo.getAggregationType();
+    if (aggregationType.equalsIgnoreCase(AggregationFunctionFactory.AggregationFunctionType.MAX.getName())
```

Comment:
extract the functiona name check into a util function (Its also used in metadatabased operator) AggregationFunctionType.isOfType(MIN, MAX, MINMAX)